src/mixins/legal-api-mixin.ts

```diff
@@ -0,0 +1,118 @@
+// Libraries
+import axios from '@/utils/axios-auth'
+import { Component, Vue } from 'vue-property-decorator'
+import { State, Action, Getter } from 'vuex-class'
+
+// Interfaces
+import { StateModelIF, ActionBindingIF, GetterIF, IncorporationFilingIF } from '@/interfaces'
+
+/**
+ * Mixin that provides the integration with the legal api.
+ */
+@Component
+export default class LegalApiMixin extends Vue {
+  // Global state
+  @State stateModel!: StateModelIF
+
+  // Global Getters
+  @Getter isTypeBcomp!: GetterIF
+  @Getter getFilingId!: number
+  @Getter getBusinessIdentifier!: string
+
+  // Store Actions
+  @Action setNameRequestState!: ActionBindingIF
+  @Action setFilingId!: ActionBindingIF
+
+  /**
+   * Method to save a draft or real filing.
+   * @param isDraft Boolean indicating if this filing is a draft
+   * @param filingId Optional filing id if this resuming an existing draft
+   */
+  async saveFiling (filing: IncorporationFilingIF, isDraft: boolean): Promise<any> {
+    try {
+      let filingId = this.getFilingId
+
+      // If have a filing id, update an existing filing
+      if (filingId && filingId > 0) {
+        this.updateFiling(isDraft, filing, filingId)
```

Comment:
Don't we need to 'await' updateFiling() (and also createFiling() below)?

Reply:
Also for getFiling() ?

Reply:
Definitely, especially when we decide we want to do something with the response.